### PR TITLE
DBA: FR9 test cases and validation

### PR DESCRIPTION
**Pull request by allenbross-wf**
#### Description of change

DBA.FR9 : Profit for the year (in the profit and loss statement) must be filled out

The year's result (fsa:ProfitLoss) must be filled in as part of the income statement.  The control only looks at instances without dimensions in or instances that only have the dimension group (ConsolidatedMember) in context. Occurrences of ""Year's result"" with other dimensions do not lift the reporting obligation."

The validation finds all facts tagged with fsa:ProfitLoss and then checks to see if there is a non-dimensionalized one or if there is one dimensionalized with cmn:ConsolidatedSoloDimension and  cmn: ConsolidatedMember

#### Steps to Test

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: arelle/plugin/validate/DBA/rules/fr.py [hunk 1]
```diff
@@ -94,6 +97,41 @@ def rule_fr7(
...
+
+    Implementation: Find any occurrence of fsa:ProfitLoss with no dimensions or with dimensions of
+    ConsolidatedSoloDimension with ConsolidatedMember.  If nothing is found, trigger an error.
+    """
+    profLossFacts = val.modelXbrl.factsByQname.get(pluginData.profitLossQn)
+    found = False
+    if profLossFacts:
+        for fact in profLossFacts:
+            if (fact.context is not None
+                    and fact.context.qnameDims.get(PluginValidationDataExtension.consolidatedSoloDimensionQn) == PluginValidationDataExtension.consolidatedMemberQn
```
Comment (aaroncameron-wk) [inline arelle/plugin/validate/DBA/rules/fr.py:122]: I'd suggest using `fact.context.dimMemberQname` so we retrieve the implied/default value and also don't rely on the context's internal data structure for dimension values.

File: arelle/plugin/validate/DBA/PluginValidationDataExtension.py [hunk 2]
```diff
@@ -6,17 +6,21 @@
...
+NAMESPACE_CMN = 'http://xbrl.dcca.dk/cmn'
 NAMESPACE_FSA = 'http://xbrl.dcca.dk/fsa'
 NAMESPACE_GSD = 'http://xbrl.dcca.dk/gsd'
 NAMESPACE_SOB = 'http://xbrl.dcca.dk/sob'
 
 
 class PluginValidationDataExtension(PluginData):
+    consolidatedSoloDimensionQn: QName = qname(f'{{{NAMESPACE_CMN}}}ConsolidatedSoloDimension')
+    consolidatedMemberQn: QName = qname(f'{{{NAMESPACE_CMN}}}ConsolidatedMember')
+    extraordinaryIncomeQn: QName = qname(f'{{{NAMESPACE_FSA}}}ExtraordinaryIncome')
```
Comment (aaroncameron-wk) [inline arelle/plugin/validate/DBA/PluginValidationDataExtension.py:18]: Nit: alphabetical sorting

File: arelle/plugin/validate/DBA/rules/fr.py [hunk 3]
```diff
@@ -94,6 +97,41 @@ def rule_fr7(
...
+                    and len(fact.context.qnameDims) == 1):
+                found = True
+            elif len(fact.context) and not len(fact.context.qnameDims):
```
Comment (aaroncameron-wk) [inline arelle/plugin/validate/DBA/rules/fr.py:125]: A `if fact.context is None` check with early exit above would make these if statements simpler.

**Review by aaroncameron-wk — APPROVED** [review]

**Pull request merged**